### PR TITLE
regression: Server isn't adding agent required properties to users created/updated through admin UI

### PR DESCRIPTION
**Pull request by murtaza98**
(cherry picked from commit 932b81a0a97fbf145bd4a97a76a3d81607f26f97)

## Issue(s)

https://rocketchat.atlassian.net/browse/OC-1194

### CODE REVIEW

**Review by KevLehman — APPROVED** [review]

**Comment by pierre-lehnen-rc** [comment]
Isn't that by design? There's an `afterCreateUser` callback as well and I see some places adding the same function to both.

File: apps/meteor/app/lib/server/functions/saveUser.js [hunk 1]
```diff
@@ -329,6 +329,12 @@ const saveNewUser = async function (userData, sendPassword) {
 		}
 	}
 
+	const userUpdated = await Users.findOneById(_id);
+	await callbacks.run('afterSaveUser', {
+		user: userUpdated,
+		oldUser: null,
+	});
+
```
Comment (ggazzo) [inline apps/meteor/app/lib/server/functions/saveUser.js:337]: this is not supposed to be an after create user?

File: apps/meteor/app/lib/server/functions/saveUser.js [hunk 2]
```diff
@@ -329,6 +329,12 @@ const saveNewUser = async function (userData, sendPassword) {
 		}
 	}
 
+	const userUpdated = await Users.findOneById(_id);
+	await callbacks.run('afterSaveUser', {
+		user: userUpdated,
+		oldUser: null,
+	});
```
Comment (ggazzo) [inline apps/meteor/app/lib/server/functions/saveUser.js:336]:
```suggestion
	await callbacks.run('afterSaveUser', {
		user: await Users.findOneById(_id),
		oldUser: null,
	});
```

**Review by ggazzo — DISMISSED** [review]
if, and I'm not saying this is the right way, but if it were the right solution, it wouldn't be a regression it wasn't just introduced, it's old behavior, it deserves a changeset and a proper commit. also i would point this pr to release and not to your own pr

**Review by pierre-lehnen-rc — APPROVED** [review]

**Review by KevLehman — APPROVED** [review]

**Pull request merged**